torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1223,6 +1276,31 @@ def test_all_reduce_result_cuda(self):
                 self.assertEqual(result, [_build_tensor(src + 1, expected_value)])
             self._barrier()
 
+        def call_dist_op(self, profiling_title_postfix, is_async, op, *args, expect_event=True, secondary_op_call=None, **kwargs):
+            op_calls = [lambda: op(*args, **kwargs)]
+            if secondary_op_call is not None:
+                op_calls.append(secondary_op_call)
+
+            with torch.autograd.profiler.profile() as prof:
+                works = [op_call() for op_call in op_calls]
+                if is_async:
+                    for work in works:
+                        work.wait()
+                        work._get_profiling_future().wait()
```

Comment:
I'm assuming that the test is flaky in the case where we remove this call? Is it okay to ship this prototype because we would need this explicit wait call from user code?

cc @pritamdamania87 - I guess we might be able to hack something but in the long term this will probably depend on future/work merge and we would implement this by adding a then callback like we do for RPC. Do you have any thoughts on what we can do currently?